darts/models/theta.py

```diff
@@ -126,3 +127,217 @@ def predict(self, n: int) -> 'TimeSeries':
+        if self.trend_mode == 'exponential':
+            theta0_in = np.exp(theta0_in)
+
+        if self.model_mode == 'additive':
+            theta_t = self.theta * new_ts.values() + (1 - self.theta) * theta0_in
+        elif self.model_mode == 'multiplicative' and (theta0_in > 0).all():
+            theta_t = (new_ts.values() ** self.theta) * (theta0_in ** (1 - self.theta))
+        else:
+            self.model_mode = 'additive'
+            theta_t = self.theta * new_ts.values() + (1 - self.theta) * theta0_in
+            logger.warn("Negative Theta line. Fallback to additive model")
+
+        # SES part of the decomposition.
+        self.model = hw.SimpleExpSmoothing(theta_t).fit()
+        theta2_in = self.model.fittedvalues
+
+        if self.model_mode == 'additive':
+            self.fitted_values = self.wses * theta2_in + self.wdrift * theta0_in
+        elif self.model_mode == 'multiplicative' and (theta2_in > 0).all():
+            self.fitted_values = theta2_in**self.wses * theta0_in**self.wdrift
+        else:
+            # Fallback to additive model
+            self.model_mode = 'additive'
+            theta_t = self.theta * new_ts.values() + (1 - self.theta) * theta0_in
+            self.model = hw.SimpleExpSmoothing(theta_t).fit()
+            theta2_in = self.model.fittedvalues
+            self.fitted_values = self.wses * theta2_in + self.wdrift * theta0_in
+            logger.warn("Negative Theta line. Fallback to additive model")
+
+        if self.is_seasonal:
+            if self.season_mode == 'additive':
+                self.fitted_values += self.seasonality.values()
+            elif self.season_mode == 'multiplicative':
+                self.fitted_values *= self.seasonality.values()
+        self.fitted_values *= self.mean
+        # takes too much time to create a time series for fitted_values
+        # self.fitted_values = TimeSeries.from_times_and_values(ts.time_index(), self.fitted_values)
+
+    def predict(self, n: int) -> 'TimeSeries':
+        super().predict(n)
+
+        # Forecast of the SES part.
+        forecast = self.model.forecast(n)
+
+        # Forecast of the Linear Regression part.
+        drift = self.drift(np.arange(self.length, self.length + n))
+        if self.trend_mode == 'exponential':
+            drift = np.exp(drift)
+
+        if self.model_mode == 'additive':
+            forecast = self.wses * forecast + self.wdrift * drift
+        elif self.model_mode == 'multiplicative':
+            forecast = forecast**self.wses * drift**self.wdrift
+        else:
+            raise_log(ValueError("model_mode cannot be {}".format(self.model_mode)))
+
+        # Re-apply the seasonal trend of the TimeSeries
+        if self.is_seasonal:
+
+            replicated_seasonality = np.tile(self.seasonality.pd_series()[-self.season_period:],
+                                             math.ceil(n / self.season_period))[:n]
+            if self.season_mode in ['multiplicative', 'mul']:
```

Comment:
for simplicity I would also pick one of them, and also an ENUM that we pass here might be helpful to avoid typos https://docs.python.org/3/library/enum.html

Reply:
To be sure I understand correctly, the different model arguments will accept an Enum value? 
I can change the theta method as well in the same manner. Thus Enum classes will be declared exterior to the class

Reply:
Yes this comment actually applies to both theta methods - instead of retyping "multiplicative"/"additive" string we can have an enum that would have 2 values - MULTIPLICATIVE or ADDITIVE and if you make a typo the IDE will tell you something is not matching (not the case for raw strings)